amundsen_application/static/js/components/common/BadgeList/tests/index.spec.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+
+import BadgeList from '../'
+import Flag from 'components/common/Flag';
+import { BadgeStyle } from 'config/config-types';
+import * as ConfigUtils from 'config/config-utils';
+import { Badge, TagType } from 'interfaces/Tags';
+
+describe('BadgeList', () => {
+  const expectedDisplayName = 'display name';
+  const expectedBadgeStyle = BadgeStyle.PRIMARY;
+  const getBadgeConfigSpy = jest.spyOn(ConfigUtils, 'getBadgeConfig');
+  getBadgeConfigSpy.mockImplementation(() => {
+    return {
+      displayName: expectedDisplayName,
+      style: expectedBadgeStyle,
+    };
+  });
+
+  describe('BadgeList function component', () => {
+    const badges: Badge[] = [
+      {
+        tag_name: 'test_1',
+        tag_type: TagType.BADGE,
+      },
+      {
+        tag_name: 'test_3',
+        tag_type: TagType.BADGE,
+      },
+    ];
+
+    const badgeList = shallow(<BadgeList badges={ badges } />);
+
+    it('renders a badge-list element', () => {
+      const container = badgeList.find('.badge-list')
+      expect(container.exists()).toBe(true);
+    });
+
+    it('renders a <Flag> for each badge in the input', () => {
+      expect(badgeList.find(Flag).length).toEqual(badges.length);
+    });
+
+    it('passes the correct props to the flag', () => {
```

Comment:
It seems the real logic we need to test is that getBadgeConfig is called and the expected values passed to the flag. I would consider replacing this test with some tests like:
```
// syntax may need to be tweaked, or tests combined however appropriate
badges.forEach((badge, index) => {
     expect(getBadgeConfigSpy).toHaveBeenCalledWith(expectedName);
})

badges.forEach((badge, index) => {
     const flag = badgeList.childAt(index);
     const flagProps = flag.props();
     expect(flagProps.text).toEqual(expectedDisplayName);
     expect(flagProps.labelStyle).toEqual(expectedBadgeStyle)
})
```